pinot-common/src/main/java/org/apache/pinot/common/lineage/LineageEntry.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.lineage;
+
+import java.util.List;
+import org.apache.pinot.spi.utils.EqualityUtils;
+
+
+/**
+ * Class to represent the lineage entry.
+ *
+ */
+public class LineageEntry {
+  private List<String> _segmentsFrom;
+  private List<String> _segmentsTo;
+  private LineageEntryState _state;
+  private long _timestamp;
+
+  public LineageEntry(List<String> segmentsFrom, List<String> segmentsTo, LineageEntryState state, long timestamp) {
+    _segmentsFrom = segmentsFrom;
+    _segmentsTo = segmentsTo;
+    _state = state;
+    _timestamp = timestamp;
+  }
+
+  public List<String> getSegmentsFrom() {
+    return _segmentsFrom;
+  }
+
+  public List<String> getSegmentsTo() {
+    return _segmentsTo;
+  }
+
+  public LineageEntryState getState() {
+    return _state;
+  }
+
+  public long getTimestamp() {
+    return _timestamp;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (EqualityUtils.isSameReference(this, o)) {
+      return true;
+    }
+    if (EqualityUtils.isNullOrNotSameClass(this, o)) {
+      return false;
+    }
+    LineageEntry that = (LineageEntry) o;
+    return EqualityUtils.isEqual(_segmentsFrom, that._segmentsFrom) && EqualityUtils.isEqual(_segmentsTo,
+        that._segmentsTo) && EqualityUtils.isEqual(_state, that._state) && EqualityUtils.isEqual(_timestamp,
+        that._timestamp);
```

Comment:
(nit) Replace with the IDE auto-generated code (use `Objects.equals()`)
```suggestion
    if (this == o) {
      return true;
    }
    if (!(o instanceof LineageEntry)) {
      return false;
    }
    LineageEntry that = (LineageEntry) o;
    return Objects.equals(_segmentsFrom, that._segmentsFrom) && Objects.equals(_segmentsTo, that._segmentsTo) && _state == that._state && _timestamp == that._timestamp;
```

Reply:
fixed